gpii/node_modules/matchMakerFramework/src/MatchMakerUtilities.js

```diff
@@ -175,6 +188,135 @@ var fluid = fluid || require("infusion"),
+
+                // if no ties, mark the current solution as "accept"
+                if (tied.length === 0) {
+                    solrecs[solutionId].disposition = "accept";
+                    solrecs[solutionId].dispositionReason = "Has a higher priority (" + priority +
+                        ") than any solution of the same type";
+                } else {
+                    // if there are ties, we "unreject" all of them, removing their disposition completely
+                    // so they can be tiebreaked by other disposal means (such as via the canopy)
+                    tied.push(solutionId);
+                    gpii.matchMakerFramework.utils.removePriorityTies(tied, solrecs);
+                }
+            }
+        }
+    };
+
+    gpii.matchMakerFramework.utils.removePriorityTies = function (tied, solrecs) {
+        fluid.each(tied, function (tiedSolutionId) {
+            delete solrecs[tiedSolutionId].priority;
+            delete solrecs[tiedSolutionId].disposition;
+            delete solrecs[tiedSolutionId].dispositionReason;
+        });
+    };
+
+    /**
+     * Rejects all solutions of a given type. This is useful for ensuring that a user does not
+     * get eg. multiple versions of screenReader launched
+     * doc: modifies the solrecs object
+     */
+    gpii.matchMakerFramework.utils.rejectFromTypes = function (types, solutionTypeMapping, solrecs, disposeReason) {
+        fluid.each(types, function (type) {
+            for (var solutionId in solutionTypeMapping[type]) {
+                var sol = solrecs[solutionId];
+                if (!sol.disposition) {
+                    sol.disposition = "reject";
+                    sol.dispositionReason = disposeReason || "Solution rejected due to its type of " + type;
+                }
+            }
+        });
+    };
+
+    gpii.matchMakerFramework.utils.parsePriorities = function (context, contextId, data, solrecs) {
+        // for each entry in the metadata block:
+        fluid.each(context.metadata, function (metadata) {
+            if (metadata.type === "priority") {
+                fluid.each(metadata.scope, function (applicationId) {
+                    // set the priority of the application if it's part of our solrecs list:
+                    var solrec = solrecs[applicationId.substr("http://registry.gpii.net/applications/".length)];
+                    if (solrec) {
+                        solrec.priority = metadata.value;
+                    }
+                });
+            }
+        });
+    };
+
+    gpii.matchMakerFramework.utils.addPriorityFromApplicationSettings = function (preferences, solrecs) {
+        fluid.each(preferences.applications, function (value, applicationId) {
+            var sol = solrecs[applicationId];
+            // If solution is available on the system, and doesn't already have higher priority
+            if (sol && (!sol.priority || sol.priority < 512)) {
```

Comment:
These magic numbers should be constants in the gpii.matchMakerFramework namespace with explanations